src/main/java/org/jenkinsci/plugins/workflow/graphanalysis/ChunkFinder.java

```diff
@@ -6,13 +6,34 @@
 import javax.annotation.Nonnull;
 
 /**
- * Used to define the start and end of a {@link FlowChunk} to split a {@link org.jenkinsci.plugins.workflow.flow.FlowExecution}
- * (For use with a {@link SimpleChunkVisitor} in the {@link ForkScanner#visitSimpleChunks(SimpleChunkVisitor, ChunkFinder)}
+ * Think of this as setting conditions to mark a region of interest in the graph of {@link FlowNode} from a {@link org.jenkinsci.plugins.workflow.flow.FlowExecution}.
+ * <p>This is used to define a linear "chunk" from the graph of FlowNodes returned by a {@link ForkScanner}, after it applies ordering.
+ * <p>This is done by invoking {@link ForkScanner#visitSimpleChunks(SimpleChunkVisitor, ChunkFinder)}.
+ * <p>Your {@link SimpleChunkVisitor} will receive callbacks about chunk boundaries on the basis of the ChunkFinder.
+ *   It isresponsible for tracking the state based on events fired
+ *
+ * <p><p><em>Common uses:</em>
+ * <ul>
+ *     <li>Find all nodes within a specific block type, such the block created by a timeout block, 'node' (executor) block, etc</li>
+ *     <li>Find all nodes between specific markers, such as labels, milestones, or steps generating an error</li>
```

Comment:
Is there any actually any use case for finding all nodes between two `milestone`s? This will not in general be a well-formed subtree.

(For labels, this only applies to the deprecated `stage` mode.)